### PR TITLE
Add entity server class

### PR DESCRIPTION
**Pull request by julien-me**
Add EntityServer class that connects to the domain server.
- [x] Add EntityServer.ts class.
- [x] Add unit test & integration test.
- [x] Update Vircadia.ts and associated integration test.
- [x] Add EntityServer section to example app, with connection status.

### CODE REVIEW

File: example/interface.html [hunk 1]
```diff
@@ -126,6 +126,14 @@ <h3>Avatar List</h3>
         </table>
     </div>
 
+    <h2>Entity Server</h2>
+    <div>
+        <div class="row">
+            <span>Status:</span>
+            <input id="EntityServerStatus" type="text" readonly="readonly" />
```
Comment (ctrlaltdavid) [inline example/interface.html:133]:
```suggestion
            <input id="entityServerStatus" type="text" readonly="readonly" />
```

File: example/interface.js [hunk 2]
```diff
@@ -8,7 +8,7 @@
 //  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
 //
 
-import { Vircadia, DomainServer, AudioMixer, AvatarMixer, MessageMixer, Uuid } from "../dist/Vircadia.js";
+import { Vircadia, DomainServer, AudioMixer, AvatarMixer, MessageMixer, Uuid, EntityServer } from "../dist/Vircadia.js";
```
Comment (ctrlaltdavid) [inline example/interface.js:11]:
```suggestion
import { Vircadia, DomainServer, AudioMixer, AvatarMixer, MessageMixer EntityServer, Uuid } from "../dist/Vircadia.js";
```
Primary items before secondary (i.e., Uuid).

File: example/interface.js [hunk 3]
```diff
@@ -348,6 +349,20 @@ import { Vircadia, DomainServer, AudioMixer, AvatarMixer, MessageMixer, Uuid } f
         return avatarMixer.myAvatar.position;
     };
 
+    // Entity Server
+    (function () {
+        entityServer = new EntityServer(contextID);
+
+        const statusText = document.getElementById("EntityServerStatus");
```
Comment (ctrlaltdavid) [inline example/interface.js:356]:
```suggestion
        const statusText = document.getElementById("entityServerStatus");
```

File: example/interface.js [hunk 4]
```diff
@@ -348,6 +349,20 @@ import { Vircadia, DomainServer, AudioMixer, AvatarMixer, MessageMixer, Uuid } f
         return avatarMixer.myAvatar.position;
     };
 
+    // Entity Server
```
Comment (ctrlaltdavid) [inline example/interface.js:352]: Move this section up to above "// Wire up mixers."

File: src/Vircadia.ts [hunk 5]
```diff
@@ -48,6 +48,7 @@ export type { AudioPositionGetter } from "./domain/audio-client/AudioClient";
 export { default as AvatarMixer, MyAvatarInterface, AvatarListInterface } from "./AvatarMixer";
 export { default as MessageMixer } from "./MessageMixer";
 export type { AssignmentClientState } from "./domain/AssignmentClient";
+export { default as EntityServer } from "./EntityServer";
```
Comment (ctrlaltdavid) [inline src/Vircadia.ts:51]:
```suggestion
export { default as EntityServer } from "./EntityServer";
export type { AssignmentClientState } from "./domain/AssignmentClient";
```

File: src/EntityServer.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+import AssignmentClient from "./domain/AssignmentClient";
+import NodeType from "./domain/networking/NodeType";
+
+/*@sdkdoc
+ *  The <code>EntityServer</code> class  provides the interface for working with entity server assignment clients.
```
Comment (ctrlaltdavid) [inline src/EntityServer.ts:15]:
```suggestion
 *  The <code>EntityServer</code> class provides the interface for working with entity server assignment clients.
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]
Nothing major except that I just noticed that the `if (nodeType === NodeType.EntityServer)` sections in DomainServer.ts's `#nodeAdded`, `#nodeActivated` and ` #nodeKilled` methods also need addressing.

**Comment by ctrlaltdavid** [comment]
`DomainServer.#nodeAdded`: Can replace the `console.warn` and `WEBRTC TODO` with a single comment:
```
// Safe landing code not implemented.
```

`DomainServer.#nodeActivated`: Can replace the entity server `console.warn` and `WEBRTC TODO` with comments:
```
// WEBRTC TODO: Address further code.

// Safe landing code not implemented.
```
We'll come back to this TODO later on in the EntityServer work.

`DomainServer.#nodeKilled`: Can replace the entity server `console.warn` and `WEBRTC TODO` with a single comment:
```
// WEBRTC TODO: Address further code.
```
We'll come back to this TODO later on in the EntityServer work.

File: example/interface.html (hunk 1 above)
Comment (julien-me) [reply example/interface.html]: Fixed.

File: example/interface.js (hunk 2 above)
Comment (julien-me) [reply example/interface.js]: Fixed.

File: example/interface.js (hunk 3 above)
Comment (julien-me) [reply example/interface.js]: Fixed.

File: example/interface.js (hunk 4 above)
Comment (julien-me) [reply example/interface.js:346]: Fixed.

File: src/Vircadia.ts (hunk 5 above)
Comment (julien-me) [reply src/Vircadia.ts]: Fixed.

File: src/EntityServer.ts (hunk 6 above)
Comment (julien-me) [reply src/EntityServer.ts]: Fixed.

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by ctrlaltdavid** [comment]
Reviewed code. Built. Linted. Ran tests.
Built and checked dev and SDK doc.
Tested example app: It connects to and disconects from the entity server the same as for the other assignment client types. It maintains a connection to the domain and a native client sees web client avatar orb.
... Looks good to me.

**Pull request merged**